Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp

```diff
@@ -1322,15 +1322,20 @@ namespace AZ
         }
 
         // All dynamic modules have been gathered at this point
-        AZ::ModuleManagerRequests::LoadModulesResult loadModuleOutcomes;
-        ModuleManagerRequestBus::BroadcastResult(loadModuleOutcomes, &ModuleManagerRequests::LoadDynamicModules, gemModules, ModuleInitializationSteps::RegisterComponentDescriptors, true);
+        for (ModuleInitializationSteps lastStepToPerform :
+             { ModuleInitializationSteps::CreateClass, ModuleInitializationSteps::RegisterComponentDescriptors })
```

Comment:
Can you do me a favor and add a comment explaining that we are loading the dynamic modules up until two phase at this point?

The first call to `LoadDynamicModules` is to ensure that all DynamicModules are loaded aand have created their AZ::Module class instances created.

The second call is perform a horizontal register step for each module component descriptors.
This allows RegisterComponentDescriptors to run when after all gems have been loaded